presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunction.java

```diff
@@ -123,12 +124,12 @@ private boolean matchesParameterAndReturnTypes(Method method, List<Type> resolve
 
         Class<?>[] methodParameterJavaTypes = method.getParameterTypes();
         for (int i = 0; i < resolvedTypes.size(); ++i) {
-            if (!methodParameterJavaTypes[i].equals(resolvedTypes.get(i).getJavaType())) {
+            if (!unwrap(methodParameterJavaTypes[i]).equals(unwrap(resolvedTypes.get(i).getJavaType()))) {
```

Comment:
Also, if this is a pre-existing issue, can you pull it out into a separate commit?